ansible_collections/arista/avd/roles/eos_designs_to_containerlab/tasks/deploy.yml

```diff
@@ -0,0 +1,237 @@
+
+- name: Create local containerlab output directories per containerlab host
+  delegate_to: localhost
+  file:
+    path: "{{ item }}"
+    state: directory
+    mode: 0775
+  loop:
+    - "{{ output_dir }}/{{ containerlab_dir_name }}/{{ inventory_hostname }}_{{ containerlab_mapping_dir_name }}"
+    - "{{ output_dir }}/{{ containerlab_dir_name }}/{{ inventory_hostname }}_{{ containerlab_config_dir_name }}"
+
+- name: Copy the needed config files per containerlab host
+  delegate_to: localhost
+  copy: src={{ eos_config_dir }}/{{ item }}.{{ avd_config_file_format }} dest={{ output_dir }}/{{ containerlab_dir_name }}/{{ inventory_hostname }}_{{ containerlab_config_dir_name }}/{{ item }}.{{ avd_config_file_format }}
+  with_items:
+    - "{{ clab_to_avd_nodes_mapping[inventory_hostname] }}"
+
+- name: Copy the needed interface mapping files per containerlab host
+  delegate_to: localhost
+  copy: src={{ containerlab_mapping_dir }}/{{ item }}.{{ avd_interface_mapping_file_format }} dest={{ output_dir }}/{{ containerlab_dir_name }}/{{ inventory_hostname }}_{{ containerlab_mapping_dir_name }}/{{ item }}.{{ avd_interface_mapping_file_format }}
+  with_items:
+    - "{{ clab_to_avd_nodes_mapping[inventory_hostname] }}"
+
+- name: Remove the temporary interface mapping directory
+  delegate_to: localhost
+  file:
+    path: "{{ containerlab_mapping_dir }}"
+    state: absent
+  run_once: true
+
+- name: Create a temporary archive of the containerlab files locally per containerlab host
+  delegate_to: localhost
+  archive:
+    path:
+      - "{{ containerlab_dir }}/{{ inventory_hostname }}*"
+    dest: "{{ containerlab_dir }}/{{ inventory_hostname }}_tmp_containerlab_archive.gz"
+
+- name: Remove containerlab directory on the containerlab host 
+  file:
+    path: "{{ ansible_facts.env.PWD }}/{{ containerlab_labname }}"
+    state: absent
+  become: yes
+
+- name: Create required directories on the containerlab host
+  file:
+    path: "{{ ansible_facts.env.PWD }}/{{ containerlab_labname }}"
+    state: directory
+    mode: 0775
+
+- name: Unarchive the containerlab files to the containerlab host
+  unarchive:
+    src: "{{ containerlab_dir }}/{{ inventory_hostname }}_tmp_containerlab_archive.gz"
+    dest: "{{ ansible_facts.env.PWD }}/{{ containerlab_labname }}"
+
+- name: Remove the temporary archive of the containerlab files locally
+  delegate_to: localhost
+  file:
+    path: "{{ containerlab_dir }}/{{ inventory_hostname }}_tmp_containerlab_archive.gz"
+    state: absent
+
+- name: Delete vxlan interfaces which will be used by the new lab
+  become: yes
+  command: "bash {{ inventory_hostname }}_vxlan_interface_delete.sh"
+  args:
+    chdir: "{{ ansible_facts.env.PWD }}/{{ containerlab_labname }}"
+  when: multinode
+  ignore_errors: yes
+
+- name: Creating run parameters for containerlab deploy
+  set_fact:
+    containerlab_cmd: "\
+    {% if containerlab_max_workers is defined %} --max-workers {{ containerlab_max_workers }}{% endif %}\
```

Comment:
please use `arista.avd.defined` to also catch the `none` case.

Reply:
changed to `arista.avd.defined`